---
title: "vizWithSCE"
author: "Kwame Forbes"
date: "`r format(Sys.Date(), '%m/%d/%Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{vizWithSCE}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

# Introduction

*vizWithSCE* is an R package that assists with the visualization of
the integration of bulk DE results tables with pre-processed scRNA-seq
datasets available on Bioconductor. vizWithSCE currently takes two
arguments, a list containing res, dds, and a SingleCellExperiment
(sce) as selected by the user and a second argument which. The which
argument is the gene to use for the violin plot, represented as a
number that corresponds to the gene with the n-th smallest 
p-value, e.g., 1 for the gene with lowest p-value, etc. 

# Run DESeq2 to obtain bulk DE results

We first load gene-summarized RNA-seq counts.

```{r eval=FALSE}
library(airway)
```

For our workflow, we used the bulk dataset "airway", which contains
four human airway smooth muscle cell lines treated with dexamethasone,
a steroid used to treat inflammation. This pre-processed dataset is
available on Bioconductor. 
Here we filter to at least 3 samples with a count of 10 or higher, and
then perform differential expression analysis.

Mike: why did you change the rownames?

```{r eval=FALSE}
library(DESeq2)
data(gse)
levels(gse$condition) <- c("untrt", "trt")
dds <- DESeqDataSet(gse, design = ~donor + condition)
keep <- rowSums(counts(dds) >= 10) >= 3
dds <- dds[keep,]
dds <- DESeq(dds)
res <- results(dds)
rownames(res) <- sub("\\..*","",rownames(res))
```

```{r echo=FALSE, results="hide"}
# the following is evaluated but hidden,
# because the above un-evaluated code chunk
# is slow to run
library(vizWithSCE)
data(dds)
dds <- DESeq2::estimateSizeFactors(dds)
data(res)
```

# Run `integrateWithSingleCell`

We run `integrateWithSingleCell` and pick number 8 from the dataset
selection as our single-cell dataset. Within PBMC there are nine
different scRNA-seq datasets. We chose the dataset pbmc4k, which
contains roughly 4 thousand cells. This pre-processed dataset is
available on Bioconductor. 

```{r eval=FALSE}
dat <- integrateWithSingleCell(res, dds)
```

```{r echo=FALSE, results="hide", message=FALSE}
# the following is evaluated but hidden,
# because the above un-evaluated code chunk
# involves user input.
library(TENxPBMCData)
library(scater)
tenx_pbmc4k <- TENxPBMCData(dataset = "pbmc4k")
sce <- tenx_pbmc4k
dat <- list(res=res, dds=dds, sce=sce)
dat$sce <- logNormCounts(dat$sce)
```

# Label cell types or clusters

Mike: what is scater used here for?

```{r eval=FALSE}
library(scater)
dat$sce <- scater::logNormCounts(dat$sce)
sce2 <- dat$sce
```

Here we convert the rownames from Ensembl to symbol, to match with the
cell annotation reference dataset, described below.

```{r eval=FALSE}
library(org.Hs.eg.db)
rownames(sce2) <- mapIds(org.Hs.eg.db, rownames(sce2), "SYMBOL", "ENSEMBL")
```

Mike: what is celldex used for here? what is SingleR used here for?

Mike: after this step what has changed in `dat`?

```{r eval=FALSE}
library(celldex)
library(SingleR)
ref <- celldex::BlueprintEncodeData()
pred <- SingleR::SingleR(test=sce2, ref=ref, labels=ref$label.main)
table(pred$labels)
colLabels(dat$sce) <- pred$labels
```

```{r echo=FALSE, results="hide"}
# the following is evaluated but hidden,
# because the above un-evaluated code chunk
# is slow to run
data(labels)
colLabels(dat$sce) <- labels
```

# Use *vizWithSCE* for visualization 

Mike: what does the following plot tell us about the gene with the 2nd
smallest p-value? Which single cells is it expressed in?

```{r viz-plot}
library(vizWithSCE)
vizWithSCE(dat, which=2)
```

# Future direction

Mike: what else would you like to do with the package? You can list
anything you like here.

# Session info

```{r}
sessionInfo()
```
