<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>vizWithSCE • vizWithSCE</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="vizWithSCE">
<meta property="og:description" content="vizWithSCE">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">vizWithSCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/vizWithSCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="vizWithSCE_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>vizWithSCE</h1>
                        <h4 class="author">Kwame Forbes</h4>
            
            <h4 class="date">04/09/2021</h4>
      
      
      <div class="hidden name"><code>vizWithSCE.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p><em>vizWithSCE</em> is an R package that assists with the visualization of the integration of bulk DE results tables with pre-processed scRNA-seq datasets available on Bioconductor. vizWithSCE currently takes two arguments, a list containing res, dds, and a SingleCellExperiment (sce) as selected by the user and a second argument which. The which argument is the gene to use for the violin plot, represented as a number that corresponds to the gene with the n-th smallest p-value, e.g., 1 for the gene with lowest p-value, etc.</p>
</div>
<div id="run-deseq2-to-obtain-bulk-de-results" class="section level1">
<h1 class="hasAnchor">
<a href="#run-deseq2-to-obtain-bulk-de-results" class="anchor"></a>Run DESeq2 to obtain bulk DE results</h1>
<p>We first load gene-summarized RNA-seq counts.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></code></pre></div>
<p>For our workflow, we used the bulk dataset “airway”, which contains four human airway smooth muscle cell lines treated with dexamethasone, a steroid used to treat inflammation. This pre-processed dataset is available on Bioconductor. Here we filter to at least 3 samples with a count of 10 or higher, and then perform differential expression analysis.</p>
<p>Mike: why did you change the rownames?</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/DESeq2">DESeq2</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">gse</span><span class="op">$</span><span class="va">condition</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"untrt"</span>, <span class="st">"trt"</span><span class="op">)</span>
<span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSet</a></span><span class="op">(</span><span class="va">gse</span>, design <span class="op">=</span> <span class="op">~</span><span class="va">donor</span> <span class="op">+</span> <span class="va">condition</span><span class="op">)</span>
<span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/counts.html">counts</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">3</span>
<span class="va">dds</span> <span class="op">&lt;-</span> <span class="va">dds</span><span class="op">[</span><span class="va">keep</span>,<span class="op">]</span>
<span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span>
<span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"\\..*"</span>,<span class="st">""</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## using 'avgTxLength' from assays(dds), correcting for library size</code></pre>
</div>
<div id="run-integratewithsinglecell" class="section level1">
<h1 class="hasAnchor">
<a href="#run-integratewithsinglecell" class="anchor"></a>Run <code>integrateWithSingleCell</code>
</h1>
<p>We run <code>integrateWithSingleCell</code> and pick number 8 from the dataset selection as our single-cell dataset. Within PBMC there are nine different scRNA-seq datasets. We chose the dataset pbmc4k, which contains roughly 4 thousand cells. This pre-processed dataset is available on Bioconductor.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/integrateWithSingleCell.html">integrateWithSingleCell</a></span><span class="op">(</span><span class="va">res</span>, <span class="va">dds</span><span class="op">)</span></code></pre></div>
</div>
<div id="label-cell-types-or-clusters" class="section level1">
<h1 class="hasAnchor">
<a href="#label-cell-types-or-clusters" class="anchor"></a>Label cell types or clusters</h1>
<p>Mike: what is scater used here for?</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/">scater</a></span><span class="op">)</span>
<span class="va">dat</span><span class="op">$</span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html">logNormCounts</a></span><span class="op">(</span><span class="va">dat</span><span class="op">$</span><span class="va">sce</span><span class="op">)</span>
<span class="va">sce2</span> <span class="op">&lt;-</span> <span class="va">dat</span><span class="op">$</span><span class="va">sce</span></code></pre></div>
<p>Here we convert the rownames from Ensembl to symbol, to match with the cell annotation reference dataset, described below.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">mapIds</span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce2</span><span class="op">)</span>, <span class="st">"SYMBOL"</span>, <span class="st">"ENSEMBL"</span><span class="op">)</span></code></pre></div>
<p>Mike: what is celldex used for here? what is SingleR used here for?</p>
<p>Mike: after this step what has changed in <code>dat</code>?</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LTLA/celldex">celldex</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LTLA/SingleR">SingleR</a></span><span class="op">)</span>
<span class="va">ref</span> <span class="op">&lt;-</span> <span class="fu">celldex</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/celldex/man/BlueprintEncodeData.html">BlueprintEncodeData</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">pred</span> <span class="op">&lt;-</span> <span class="fu">SingleR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SingleR/man/SingleR.html">SingleR</a></span><span class="op">(</span>test<span class="op">=</span><span class="va">sce2</span>, ref<span class="op">=</span><span class="va">ref</span>, labels<span class="op">=</span><span class="va">ref</span><span class="op">$</span><span class="va">label.main</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">pred</span><span class="op">$</span><span class="va">labels</span><span class="op">)</span>
<span class="fu">colLabels</span><span class="op">(</span><span class="va">dat</span><span class="op">$</span><span class="va">sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">pred</span><span class="op">$</span><span class="va">labels</span></code></pre></div>
</div>
<div id="use-vizwithsce-for-visualization" class="section level1">
<h1 class="hasAnchor">
<a href="#use-vizwithsce-for-visualization" class="anchor"></a>Use <em>vizWithSCE</em> for visualization</h1>
<p>Mike: what does the following plot tell us about the gene with the 2nd smallest p-value? Which single cells is it expressed in?</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">vizWithSCE</span><span class="op">)</span>
<span class="fu"><a href="../reference/vizWithSCE.html">vizWithSCE</a></span><span class="op">(</span><span class="va">dat</span>, which<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="vizWithSCE_files/figure-html/viz-plot-1.png" width="700"></p>
</div>
<div id="future-direction" class="section level1">
<h1 class="hasAnchor">
<a href="#future-direction" class="anchor"></a>Future direction</h1>
<p>Mike: what else would you like to do with the package? You can list anything you like here.</p>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.4 (2021-02-15)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Big Sur 10.16
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] scater_1.18.6               ggplot2_3.3.3              
##  [3] TENxPBMCData_1.8.0          HDF5Array_1.18.1           
##  [5] rhdf5_2.34.0                DelayedArray_0.16.2        
##  [7] Matrix_1.3-2                SingleCellExperiment_1.12.0
##  [9] SummarizedExperiment_1.20.0 Biobase_2.50.0             
## [11] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
## [13] IRanges_2.24.1              S4Vectors_0.28.1           
## [15] BiocGenerics_0.36.0         MatrixGenerics_1.2.1       
## [17] matrixStats_0.58.0          vizWithSCE_0.0.4           
## 
## loaded via a namespace (and not attached):
##   [1] ggbeeswarm_0.6.0              colorspace_2.0-0             
##   [3] ellipsis_0.3.1                rprojroot_2.0.2              
##   [5] scuttle_1.0.4                 XVector_0.30.0               
##   [7] BiocNeighbors_1.8.2           fs_1.5.0                     
##   [9] farver_2.1.0                  bit64_4.0.5                  
##  [11] interactiveDisplayBase_1.28.0 AnnotationDbi_1.52.0         
##  [13] fansi_0.4.2                   splines_4.0.4                
##  [15] sparseMatrixStats_1.2.1       cachem_1.0.4                 
##  [17] geneplotter_1.68.0            knitr_1.31                   
##  [19] polyclip_1.10-0               jsonlite_1.7.2               
##  [21] annotate_1.68.0               dbplyr_2.1.0                 
##  [23] ggforce_0.3.3                 shiny_1.6.0                  
##  [25] BiocManager_1.30.10           compiler_4.0.4               
##  [27] httr_1.4.2                    assertthat_0.2.1             
##  [29] fastmap_1.1.0                 BiocSingular_1.6.0           
##  [31] later_1.1.0.1                 tweenr_1.0.2                 
##  [33] htmltools_0.5.1.1             tools_4.0.4                  
##  [35] rsvd_1.0.3                    gtable_0.3.0                 
##  [37] glue_1.4.2                    GenomeInfoDbData_1.2.4       
##  [39] dplyr_1.0.5                   rappdirs_0.3.3               
##  [41] Rcpp_1.0.6                    jquerylib_0.1.3              
##  [43] pkgdown_1.6.1                 vctrs_0.3.6                  
##  [45] rhdf5filters_1.2.0            ExperimentHub_1.16.0         
##  [47] DelayedMatrixStats_1.12.3     xfun_0.22                    
##  [49] stringr_1.4.0                 beachmat_2.6.4               
##  [51] irlba_2.3.3                   mime_0.10                    
##  [53] lifecycle_1.0.0               XML_3.99-0.5                 
##  [55] AnnotationHub_2.22.0          zlibbioc_1.36.0              
##  [57] MASS_7.3-53                   scales_1.1.1                 
##  [59] ragg_1.1.2                    promises_1.2.0.1             
##  [61] RColorBrewer_1.1-2            yaml_2.2.1                   
##  [63] curl_4.3                      gridExtra_2.3                
##  [65] memoise_2.0.0                 sass_0.3.1                   
##  [67] stringi_1.5.3                 RSQLite_2.2.3                
##  [69] highr_0.8                     BiocVersion_3.12.0           
##  [71] genefilter_1.72.1             desc_1.3.0                   
##  [73] BiocParallel_1.24.1           rlang_0.4.10                 
##  [75] pkgconfig_2.0.3               systemfonts_1.0.1            
##  [77] bitops_1.0-6                  evaluate_0.14                
##  [79] lattice_0.20-41               purrr_0.3.4                  
##  [81] Rhdf5lib_1.12.1               labeling_0.4.2               
##  [83] bit_4.0.4                     tidyselect_1.1.0             
##  [85] magrittr_2.0.1                DESeq2_1.30.1                
##  [87] R6_2.5.0                      generics_0.1.0               
##  [89] DBI_1.1.1                     pillar_1.5.0                 
##  [91] withr_2.4.1                   survival_3.2-7               
##  [93] RCurl_1.98-1.2                tibble_3.1.0                 
##  [95] crayon_1.4.1                  utf8_1.1.4                   
##  [97] BiocFileCache_1.14.0          rmarkdown_2.7                
##  [99] viridis_0.5.1                 locfit_1.5-9.4               
## [101] grid_4.0.4                    blob_1.2.1                   
## [103] digest_0.6.27                 xtable_1.8-4                 
## [105] httpuv_1.5.5                  textshaping_0.3.3            
## [107] munsell_0.5.0                 viridisLite_0.3.0            
## [109] beeswarm_0.3.1                vipor_0.4.5                  
## [111] bslib_0.2.4</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Kwame Forbes, Michael Love.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
